homeassistant/components/vevor_heater/coordinator.py

```diff
@@ -0,0 +1,69 @@
+"""Data update coordinators for Vevor BLE Heater integration."""
+
+import logging
+from typing import Optional
+
+from bleak import BLEDevice
+import bluetooth_data_tools
+from vevor_heater_ble.heater import VevorDevice
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_UPDATE_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class VevorHeaterUpdateCoordinator(DataUpdateCoordinator[VevorDevice]):
+    """Update coordinator for Vevor BLE Heaters."""
+
+    def __init__(self, hass: HomeAssistant, device: BLEDevice) -> None:
+        """Initialize the coordinator for the given BLE device. Assumes the device corresponds to a Vevor Heater."""
+        self._ble_device = device
+        address = self._ble_device.address
+        name = self._ble_device.name or bluetooth_data_tools.short_address(address)
+        self._vevor_device = VevorDevice(
+            address=address,
+            name=f"Vevor {name}",
+        )
+        super().__init__(
+            hass, logger=_LOGGER, name=DOMAIN, update_interval=DEFAULT_UPDATE_INTERVAL
+        )
+
+    async def _async_update_data(self):
+        """Attempt to update the sensor data."""
+        try:
+            await self._vevor_device.refresh_status(self._ble_device)
+        except Exception as err:
```

Comment:
Please only trap expected exceptions here. We want unknown or unexpected ones to still raise